### PR TITLE
Space to depth - refactor reference implementation

### PR DESCRIPTION
**Pull request by pelszkow**
### Details:
 - *extract SpaceToDepth implementation from op class to reference file*
 - *move backend test from fused_op.in.cpp to space_to_depth.in.cpp*
 - *drop needless reshapes from reference implementation (reshapes doesn't affect elements in tensor, shape was only changed)*
 - *add serialization test*

### Tickets:
 - *44634*

### CODE REVIEW

File: inference-engine/tests/functional/inference_engine/serialization/single_layer/space_to_depth.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+#include <ngraph/opsets/opset3.hpp>
+
+#include "common_test_utils/test_constants.hpp"
+
+using namespace LayerTestsDefinitions;
+using namespace ngraph::opset3;
+
+namespace {
+TEST_P(SpaceToDepthLayerTest, Serialize) {
```
Comment (ilyachur) [inline inference-engine/tests/functional/inference_engine/serialization/single_layer/space_to_depth.cpp:15]: Could you add SLT?

Comment (pelszkow) [reply inference-engine/tests/functional/inference_engine/serialization/single_layer/space_to_depth.cpp:15]: More SLT than that: 
- https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/cpu/single_layer_tests/space_to_depth.cpp
- https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/space_to_depth.cpp

Comment (ilyachur) [reply inference-engine/tests/functional/inference_engine/serialization/single_layer/space_to_depth.cpp:15]: Don't we need to extend these tests?

**Review by jdanieck — APPROVED** [review]

**Comment by ggalieroc** [comment]
Please add the op to this [list](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/ie_test_utils/functional_test_utils/layer_tests_summary/utils/constants.py)

File: inference-engine/tests/functional/inference_engine/serialization/single_layer/space_to_depth.cpp (hunk 1 above)
Comment (pelszkow) [reply inference-engine/tests/functional/inference_engine/serialization/single_layer/space_to_depth.cpp:15]: IMHO we got decent amount of SLT for SpaceToDetph.

**Review by ggalieroc — APPROVED** [review]
LGTM. Please don't forget to address this [comment](https://github.com/openvinotoolkit/openvino/pull/6374#issuecomment-872121234)

**Review by ilyachur — APPROVED** [review]

**Pull request merged**